### ISSUE TITLE
Error executing a module right after creation

### ISSUE THREAD

**Issue by clun** [jhipster/generator-jhipster-module#47]
Hi team,

Not sure if it is on my hand but here is what i do and lead to issue

**Versions:**
JHIPSTER: `6.4.1`
JHipster Module Generator! `v2.5.0`

**Scenario to reproduce:**
- Installing the generator module `npm install -g generator-jhipster-module`
- Generating a new module aaa (mkdir, cd..) and `yo jhipster-module` => SUCCESS

```
##### USAGE #####
To begin to work:
- launch: npm install or yarn install
- link: npm link or yarn link
- test your module in a JHipster project: 
    - go into your JHipster project
    - link to your module: npm link generator-jhipster-aaa or yarn link generator-jhipster-aaa
    - launch your module: yo jhipster-aaa
- then, come back here, and begin to code!
```

- Launch `npm install`
- Link `npm install`
- Create a new jhipster App (mkdir, cd, jhipster) ==> SUCCESS
```
Running sanity checks on your system
✔ No .bowerrc file in home directory
✔ Global configuration file is valid
✔ NODE_PATH matches the npm root
✔ No .yo-rc.json file in home directory
✔ Node.js version
✔ npm version
✔ yo version
Everything looks all right!
```

- In the generated App I link my generator `npm link generator-jhipster-aaa`

```
cedricklunven@clunhost:~/dev/WORKSPACES/JHIPSTER/app1> npm link generator-jhipster-aaa
/Users/cedricklunven/dev/WORKSPACES/JHIPSTER/app1/node_modules/generator-jhipster-aaa -> /usr/local/lib/node_modules/generator-jhipster-aaa -> /Users/cedricklunven/dev/WORKSPACES/JHIPSTER/generator-jhipster-aaa
```

Then when I : `yo jhipster-aaa` this is the error message I got :

```
cedricklunven@clunhost:~/dev/WORKSPACES/JHIPSTER/app1> yo jhipster-aaa
events.js:183
      throw er; // Unhandled 'error' event
      ^
TypeError: this.getJhipsterAppConfig is not a function
    at module.exports.readConfig (/Users/cedricklunven/dev/WORKSPACES/JHIPSTER/generator-jhipster-aaa/generators/app/index.js:16:47)
    at Object.<anonymous> (/Users/cedricklunven/dev/WORKSPACES/JHIPSTER/generator-jhipster-aaa/node_modules/generator-jhipster/node_modules/yeoman-generator/lib/index.js:424:27)
    at /Users/cedricklunven/dev/WORKSPACES/JHIPSTER/generator-jhipster-aaa/node_modules/run-async/index.js:25:25
    at new Promise (<anonymous>)
    at /Users/cedricklunven/dev/WORKSPACES/JHIPSTER/generator-jhipster-aaa/node_modules/run-async/index.js:24:19
    at self.env.runLoop.add.completed (/Users/cedricklunven/dev/WORKSPACES/JHIPSTER/generator-jhipster-aaa/node_modules/generator-jhipster/node_modules/yeoman-generator/lib/index.js:425:13)
    at runCallback (timers.js:794:20)
    at tryOnImmediate (timers.js:752:5)
    at processImmediate [as _immediateCallback] (timers.js:729:5)
```

Could you help me on this one ?
Thank you

**Comment by pascalgrimaud** [comment]
Hi @clun !

Just tried it, and everything works fine for me.
The only difference is:
- I'm using generator-jhipster master branch, with npm link
- I'm using generator-jhipster-module, master branch with npm link

Here my logs from the module:

```
22:15:40 in ~/tmp/generator-jhipster-ippon 
➜ npm link generator-jhipster-module
/home/pgrimaud/tmp/generator-jhipster-ippon/node_modules/generator-jhipster-module -> /home/pgrimaud/.npm-global/lib/node_modules/generator-jhipster-module -> /home/pgrimaud/projects/jhipster/generator-jhipster-module

22:16:15 in ~/tmp/generator-jhipster-ippon 
➜ yo jhipster-module     

               Three::Modules
           for:::the::Elven-Kings
        under:the:sky,:Seven:for:the
      Dwarf-Lords::in::their::halls:of
     stone,:Nine             for:Mortal
    :::Men:::    ________     doomed::to
  die.:One    _,-'...:... `-.    for:::the
 ::Dark::   ,- .:::::::::::. `. Hipster::on
 his:dark ,'  .:::::::::::::.  `.  :throne:
In:::the /    :::: Java :::::    \  Land::of
JHipster \    ::: Hipster :::    /  :where::
 ::the::: '.  ':::::::::::::'  ,'  Shadows:
  lie::One  `. ``:::::::::'' ,'  Module::to
  ::rule::    `-._```:'''_,-'     ::them::
   all,::One      `-----'       Module::to
    ::find:::                  them,:One
     Module:::::to          bring::them
       all::and::in:the:darkness:bind
         them:In:the:Land:of:JHipster
            where:::the::Shadows
                 :::lie.:::

         http://www.jhipster.tech

Welcome to the JHipster Module Generator! v2.5.0

? What is the base name of your module? ippon
? Give a description of your module 
? Do you want to enable hooks for your module from JHipster generator? No, This is a stand alone module
? What is your GitHub username? pascalgrimaud
? Who are you? Firstname Lastname Pascal Grimaud
? Your email? 
? Your home page url? 
? Do you want to add a license? Apache License 2.0
   create .prettierrc
   create .prettierignore
   create .editorconfig
   create .eslintignore
   create .gitattributes
   create .gitignore
   create LICENSE
   create generators/app/templates/dummy.txt
   create package.json
   create .eslintrc.json
   create .travis.yml
   create README.md
   create test/app.spec.js
   create generators/app/index.js

##### USAGE #####
To begin to work:
- launch: npm install or yarn install
- link: npm link or yarn link
- test your module in a JHipster project: 
    - go into your JHipster project
    - link to your module: npm link generator-jhipster-ippon or yarn link generator-jhipster-ippon
    - launch your module: yo jhipster-ippon
- then, come back here, and begin to code!


22:16:51 in ~/tmp/generator-jhipster-ippon took 31s 
➜ npm install && npm link           

> husky@3.0.7 install /home/pgrimaud/tmp/generator-jhipster-ippon/node_modules/husky
> node husky install

husky > Setting up git hooks
Command failed: git rev-parse --show-toplevel --absolute-git-dir
fatal: ni ceci ni aucun de ses répertoires parents n'est un dépôt git : .git
husky > Failed to install

> core-js@3.3.2 postinstall /home/pgrimaud/tmp/generator-jhipster-ippon/node_modules/core-js
> node postinstall || echo "ignore"


> spawn-sync@1.0.15 postinstall /home/pgrimaud/tmp/generator-jhipster-ippon/node_modules/spawn-sync
> node postinstall


> yo@3.1.0 postinstall /home/pgrimaud/tmp/generator-jhipster-ippon/node_modules/yo
> yodoctor


Yeoman Doctor
Running sanity checks on your system

✔ No .bowerrc file in home directory
✔ Global configuration file is valid
✔ NODE_PATH matches the npm root
✔ No .yo-rc.json file in home directory
✔ Node.js version
✔ npm version
✔ yo version

Everything looks all right!

> husky@3.0.7 postinstall /home/pgrimaud/tmp/generator-jhipster-ippon/node_modules/husky
> opencollective-postinstall || exit 0

Thank you for using husky!
If you rely on this package, please consider supporting our open collective:
> https://opencollective.com/husky/donate

npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN eslint-config-airbnb-base@13.0.0 requires a peer of eslint@^4.19.1 but none is installed. You must install peer dependencies yourself.

added 1038 packages from 1197 contributors, removed 1 package and audited 12158 packages in 19.963s
found 0 vulnerabilities

npm WARN eslint-config-airbnb-base@13.0.0 requires a peer of eslint@^4.19.1 but none is installed. You must install peer dependencies yourself.

audited 12158 packages in 3.598s
found 0 vulnerabilities

/home/pgrimaud/.npm-global/lib/node_modules/generator-jhipster-ippon -> /home/pgrimaud/tmp/generator-jhipster-ippon
```

Here, when I applied the module into an existing project:

```
22:16:19 in ~/tmp/17-ngx on  master [!+] took 1m 5s 
➜ npm link generator-jhipster-ippon
/home/pgrimaud/tmp/17-ngx/node_modules/generator-jhipster-ippon -> /home/pgrimaud/.npm-global/lib/node_modules/generator-jhipster-ippon -> /home/pgrimaud/tmp/generator-jhipster-ippon

22:17:34 in ~/tmp/17-ngx on  master [!+] 
➜ yo jhipster-ippon 

     _-----_     
    |       |    ╭──────────────────────────────────────────╮
    |--(o)--|    │ Update available: 3.1.0 (current: 2.0.5) │
   `---------´   │     Run npm install -g yo to update.     │
    ( _´U`_ )    ╰──────────────────────────────────────────╯
    /___A___\   /
     |  ~  |     
   __'.___.'__   
 ´   `  |° ´ Y ` 



        ██╗ ██╗   ██╗ ████████╗ ███████╗   ██████╗ ████████╗ ████████╗ ███████╗
        ██║ ██║   ██║ ╚══██╔══╝ ██╔═══██╗ ██╔════╝ ╚══██╔══╝ ██╔═════╝ ██╔═══██╗
        ██║ ████████║    ██║    ███████╔╝ ╚█████╗     ██║    ██████╗   ███████╔╝
  ██╗   ██║ ██╔═══██║    ██║    ██╔════╝   ╚═══██╗    ██║    ██╔═══╝   ██╔══██║
  ╚██████╔╝ ██║   ██║ ████████╗ ██║       ██████╔╝    ██║    ████████╗ ██║  ╚██╗
   ╚═════╝  ╚═╝   ╚═╝ ╚═══════╝ ╚═╝       ╚═════╝     ╚═╝    ╚═══════╝ ╚═╝   ╚═╝

                            https://www.jhipster.tech

Welcome to JHipster v6.4.1
Application files will be generated in folder: /home/pgrimaud/tmp/17-ngx
 _______________________________________________________________________________________________________________

  Documentation for creating an application is at https://www.jhipster.tech/creating-an-app/
  If you find JHipster useful, consider sponsoring the project at https://opencollective.com/generator-jhipster
 _______________________________________________________________________________________________________________


Welcome to the JHipster ippon generator! v0.0.0

? Please put something hello world!

--- some config read from config ---
baseName=jhipster
packageName=com.mycompany.myapp
clientFramework=angularX
clientPackageManager=npm
buildTool=maven

--- some function ---
angularAppName=jhipsterApp

--- some const ---
javaDir=src/main/java/com/mycompany/myapp/
resourceDir=src/main/resources/
webappDir=src/main/webapp/

--- variables from questions ---

message=hello world!
------

   create dummy-angularX.txt
   create dummy-maven.txt


I'm all done. Running npm install for you to install the required dependencies. If this fails, try running the command yourself.


npm WARN @angular/core@8.2.4 requires a peer of zone.js@~0.9.1 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.3.1 requires a peer of jquery@1.9.1 - 3 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.3.1 requires a peer of popper.js@^1.14.7 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

removed 1 package and audited 909582 packages in 10.894s
found 3 moderate severity vulnerabilities
  run `npm audit fix` to fix them, or `npm audit` for details
End of ippon generator
```

**Comment by pascalgrimaud** [comment]
Ok, understood.
The generator-jhipster-module v2.5.0 (released 24 days ago) is not compatible with JHipster v6 project.
See this PR https://github.com/jhipster/generator-jhipster-module/pull/31 which was merged 16 days ago.

A new release is needed.
cc @murdos : I let you do the release, unless you want me to do it

**Comment by clun** [comment]
Thank you so much @pascalgrimaud, you help even if it is late ! 

I can give a try using masters for tonight and will update `generator-jhipster-module` later when you will do the PR another day.

*Context:* tests to generate DAO with latest DataStax drivers for Cassandra (new mappers etc...) let's see what I can do. Would like also to generate JDL based on existing DB.

Thank you again !

**Comment by pascalgrimaud** [comment]
@clun : you're welcome. Don't hesitate to tell me if you need a release.
Then, maybe include it directly to generator-jhipster ? As you are the stream lead, you'll decide :)

**Comment by murdos** [comment]
Should be fixed now with the 2.6.0 release 😉
